### PR TITLE
Bioin 1515 filter by freec segments

### PR DESCRIPTION
**Pull request by TammushB**

### CODE REVIEW

File: test/system/test_annotate_FREEC_segments.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+from test import get_resource_dir, test_dir
+from ugvc.somatic_cnv import annotate_FREEC_segments
+
+class TestAnnotateFREECSegments:
+    inputs_dir = get_resource_dir(__file__)
+
+    def test_annotate_FREEC_segments(self, tmpdir):
+        input_segments_file = pjoin(self.inputs_dir, "in_segments.txt")
+        gain_cutoff = '1.03'
```
Comment (ilyasoifer) [inline test/system/test_annotate_FREEC_segments.py]: Int please

File: ugvc/__main__.py [hunk 2]
```diff
@@ -55,7 +55,10 @@
 from ugvc.pipelines.sec import assess_sec_concordance, correct_systematic_errors, sec_training, sec_validation
 from ugvc.pipelines.vcfbed import annotate_contig, calibrate_bridging_snvs, intersect_bed_regions
 from ugvc.scripts import sorter_to_h5
-from ugvc.somatic_cnv import bicseq2_post_processing
```
Comment (ilyasoifer) [inline ugvc/__main__.py:58]: Can you merge cnv and somatic_cnv please? There is no real reason for the split, right?

File: ugvc/somatic_cnv/annotate_FREEC_segments.py
Comment (ilyasoifer) [inline ugvc/somatic_cnv/annotate_FREEC_segments.py:1]: We tend to place code like this into `pipelines`

**Review by ilyasoifer — COMMENTED** [review]
Nice! Just few small comments

File: ugvc/__main__.py (hunk 2 above)
Comment (TammushB) [reply ugvc/__main__.py:58]: moved to ugbio-cnv

File: ugvc/somatic_cnv/annotate_FREEC_segments.py
Comment (TammushB) [reply ugvc/somatic_cnv/annotate_FREEC_segments.py:1]: done

File: ugvc/pipelines/cnv/annotate_FREEC_segments.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    )
+
+    args = parser.parse_args(argv[1:])
+    logger.setLevel(getattr(logging, args.verbosity))
+
+    df_segments = pd.read_csv(args.input_segments_file,sep='\t')
+    
+    gain_cutoff = args.gain_cutoff
+    loss_cutoff = args.loss_cutoff
+    df_segments['alteration']=df_segments['median_ratio'].apply(lambda x: 'gain' if x >= gain_cutoff else ('loss' if ((x <= loss_cutoff) & (x>-1)) else 'neutral'))
```
Comment (gilhornung) [inline ugvc/pipelines/cnv/annotate_FREEC_segments.py]: You can use `np.where` in order to write this a bit more clearly:
https://www.statology.org/case-statement-pandas/

**Review by gilhornung — APPROVED** [review]
Looks good

File: ugvc/pipelines/cnv/annotate_FREEC_segments.py (hunk 3 above)
Comment (gilhornung) [reply ugvc/pipelines/cnv/annotate_FREEC_segments.py]: or `np.select`

**Pull request merged**